From dfe75b670def5dd88d238d0687c642ede8e596c1 Mon Sep 17 00:00:00 2001
From: Kevin Chen <kevin_chen@aspeedtech.com>
Date: Thu, 21 Aug 2025 10:33:32 +0800
Subject: [PATCH 09/32] misc: aspeed_edaf_bridge: refactor variable usage

Refactors variable usage in the aspeed_edaf_bridge driver to enhance
clarity and maintainability. Replaces generic variable 'cfg' with
more descriptive names ('scu', 'edaf', 'espi') to improve readability.

Aligns variable naming with specific hardware register contexts, reducing
potential confusion and making the code easier to understand and maintain.

Change-Id: Ia3252488e25fabcff593d09b54a9ecd6d569c668
Signed-off-by: Kevin Chen <kevin_chen@aspeedtech.com>
(cherry picked from commit c04e20dbad0a5324f2e3ea4ad47e5d155a0d25b2)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 drivers/misc/aspeed_edaf_bridge.c | 76 +++++++++++++++----------------
 1 file changed, 38 insertions(+), 38 deletions(-)

diff --git a/drivers/misc/aspeed_edaf_bridge.c b/drivers/misc/aspeed_edaf_bridge.c
index ed64777007..bae0f7bf42 100644
--- a/drivers/misc/aspeed_edaf_bridge.c
+++ b/drivers/misc/aspeed_edaf_bridge.c
@@ -58,7 +58,7 @@ static int aspeed_edaf_bridge_probe(struct udevice *dev)
 	void *scu_regs, *espi_regs, *virt;
 	ofnode node, scu1_node, mem_node;
 	uint64_t cbase, mbase, espibase;
-	uint32_t cfg, phandle;
+	uint32_t scu, edaf, espi, phandle;
 	struct resource res;
 	int rc;
 
@@ -92,23 +92,23 @@ static int aspeed_edaf_bridge_probe(struct udevice *dev)
 		return -ENODEV;
 	}
 
-	cfg = readl(scu_regs + SCU1_HOST_CONF_1);
-	if (cfg & SCU1_HOST_CONF_LPC_EN) {
+	scu = readl(scu_regs + SCU1_HOST_CONF_1);
+	if (scu & SCU1_HOST_CONF_LPC_EN) {
 		printf("Host Configuration1: eSPI0 eDAF is not valid\n");
 		return -EINVAL;
 	}
-	cfg |= SCU1_HOST_CONF_EDAF_EN;
-	writel(cfg, scu_regs + SCU1_HOST_CONF_1);
+	scu |= SCU1_HOST_CONF_EDAF_EN;
+	writel(scu, scu_regs + SCU1_HOST_CONF_1);
 
-	cfg = readl(scu_regs + SCU1_HOST_CONF_2);
-	if (cfg & SCU1_HOST_CONF_LPC_EN) {
+	scu = readl(scu_regs + SCU1_HOST_CONF_2);
+	if (scu & SCU1_HOST_CONF_LPC_EN) {
 		printf("Host Configuration2: eSPI1 eDAF is not valid\n");
 		return -EINVAL;
 	}
-	cfg |= SCU1_HOST_CONF_EDAF_EN;
-	writel(cfg, scu_regs + SCU1_HOST_CONF_2);
+	scu |= SCU1_HOST_CONF_EDAF_EN;
+	writel(scu, scu_regs + SCU1_HOST_CONF_2);
 
-	cfg = readl(edaf_bridge_regs + EDAF_BDGE_CFG);
+	edaf = readl(edaf_bridge_regs + EDAF_BDGE_CFG);
 	if (ofnode_read_bool(node, "edaf-ddr-mode")) {
 		rc = ofnode_read_u32(node, "mem-base", &phandle);
 		if (rc) {
@@ -135,66 +135,66 @@ static int aspeed_edaf_bridge_probe(struct udevice *dev)
 			return -ENOMEM;
 		}
 
-		cfg &= ~EDAF_BDGE_CFG_CMD_EN;
+		edaf &= ~EDAF_BDGE_CFG_CMD_EN;
 	} else {
 		rc = ofnode_read_u64(node, "mem-base", &mbase);
 		if (rc)
 			printf("cannot get eDAF memory region\n");
 
-		cfg |= EDAF_BDGE_CFG_CMD_EN;
+		edaf |= EDAF_BDGE_CFG_CMD_EN;
 	}
 	writel(mbase, edaf_bridge_regs + EDAF_BDGE_MBASE);
-	writel(cfg, edaf_bridge_regs + EDAF_BDGE_CFG);
+	writel(edaf, edaf_bridge_regs + EDAF_BDGE_CFG);
 
-	cfg = readl(edaf_bridge_regs + EDAF_BDGE_MISC);
-	cfg &= ~EDAF_BDGE_MISC_MBASE_H;
-	cfg |= FIELD_PREP(EDAF_BDGE_MISC_MBASE_H, mbase >> 32);
-	writel(cfg, edaf_bridge_regs + EDAF_BDGE_MISC);
+	edaf = readl(edaf_bridge_regs + EDAF_BDGE_MISC);
+	edaf &= ~EDAF_BDGE_MISC_MBASE_H;
+	edaf |= FIELD_PREP(EDAF_BDGE_MISC_MBASE_H, mbase >> 32);
+	writel(edaf, edaf_bridge_regs + EDAF_BDGE_MISC);
 
 	rc = ofnode_read_u64(node, "ctl-base", &cbase);
 	if (!rc) {
 		writel(cbase, edaf_bridge_regs + EDAF_BDGE_CBASE);
 
-		cfg = readl(edaf_bridge_regs + EDAF_BDGE_MISC);
-		cfg &= ~EDAF_BDGE_MISC_CBASE_H;
-		cfg |= FIELD_PREP(EDAF_BDGE_MISC_CBASE_H, cbase >> 32);
-		writel(cfg, edaf_bridge_regs + EDAF_BDGE_MISC);
+		edaf = readl(edaf_bridge_regs + EDAF_BDGE_MISC);
+		edaf &= ~EDAF_BDGE_MISC_CBASE_H;
+		edaf |= FIELD_PREP(EDAF_BDGE_MISC_CBASE_H, cbase >> 32);
+		writel(edaf, edaf_bridge_regs + EDAF_BDGE_MISC);
 	}
 
-	rc = ofnode_read_u32(node, "cmd-read", &cfg);
+	rc = ofnode_read_u32(node, "cmd-read", &edaf);
 	if (!rc)
-		writel(cfg, edaf_bridge_regs + EDAF_BDGE_CMD_READ);
+		writel(edaf, edaf_bridge_regs + EDAF_BDGE_CMD_READ);
 
-	rc = ofnode_read_u32(node, "cmd-write-enable", &cfg);
+	rc = ofnode_read_u32(node, "cmd-write-enable", &edaf);
 	if (!rc)
-		writel(cfg, edaf_bridge_regs + EDAF_BDGE_CMD_WRITE_EN);
+		writel(edaf, edaf_bridge_regs + EDAF_BDGE_CMD_WRITE_EN);
 
-	rc = ofnode_read_u32(node, "cmd-write", &cfg);
+	rc = ofnode_read_u32(node, "cmd-write", &edaf);
 	if (!rc)
-		writel(cfg, edaf_bridge_regs + EDAF_BDGE_CMD_WRITE);
+		writel(edaf, edaf_bridge_regs + EDAF_BDGE_CMD_WRITE);
 
-	rc = ofnode_read_u32(node, "cmd-read-status", &cfg);
+	rc = ofnode_read_u32(node, "cmd-read-status", &edaf);
 	if (!rc)
-		writel(cfg, edaf_bridge_regs + EDAF_BDGE_CMD_READ_STS);
+		writel(edaf, edaf_bridge_regs + EDAF_BDGE_CMD_READ_STS);
 
-	rc = ofnode_read_u32(node, "cmd-erase-4k", &cfg);
+	rc = ofnode_read_u32(node, "cmd-erase-4k", &edaf);
 	if (!rc)
-		writel(cfg, edaf_bridge_regs + EDAF_BDGE_CMD_ERASE_4K);
+		writel(edaf, edaf_bridge_regs + EDAF_BDGE_CMD_ERASE_4K);
 
-	rc = ofnode_read_u32(node, "cmd-erase-32k", &cfg);
+	rc = ofnode_read_u32(node, "cmd-erase-32k", &edaf);
 	if (!rc)
-		writel(cfg, edaf_bridge_regs + EDAF_BDGE_CMD_ERASE_32K);
+		writel(edaf, edaf_bridge_regs + EDAF_BDGE_CMD_ERASE_32K);
 
-	rc = ofnode_read_u32(node, "cmd-erase-64k", &cfg);
+	rc = ofnode_read_u32(node, "cmd-erase-64k", &edaf);
 	if (!rc)
-		writel(cfg, edaf_bridge_regs + EDAF_BDGE_CMD_ERASE_64K);
+		writel(edaf, edaf_bridge_regs + EDAF_BDGE_CMD_ERASE_64K);
 
 	rc = ofnode_read_u64(node, "espi-base", &espibase);
 	if (!rc) {
 		espi_regs = (void *)(uintptr_t)espibase;
-		cfg = readl(espi_regs + ESPI_CHN3_CTL);
-		cfg |= ESPI_CHN3_SW_READY;
-		writel(cfg, espi_regs + ESPI_CHN3_CTL);
+		espi = readl(espi_regs + ESPI_CHN3_CTL);
+		espi |= ESPI_CHN3_SW_READY;
+		writel(espi, espi_regs + ESPI_CHN3_CTL);
 	}
 
 	return 0;
-- 
2.34.1

